[2025-03-06T16:12:25.137+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-03-06T16:12:25.163+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_flow.load_exchangeRate_python_job manual__2025-03-06T16:12:20.170772+00:00 [queued]>
[2025-03-06T16:12:25.175+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_flow.load_exchangeRate_python_job manual__2025-03-06T16:12:20.170772+00:00 [queued]>
[2025-03-06T16:12:25.178+0000] {taskinstance.py:2867} INFO - Starting attempt 1 of 1
[2025-03-06T16:12:25.192+0000] {taskinstance.py:2890} INFO - Executing <Task(SparkSubmitOperator): load_exchangeRate_python_job> on 2025-03-06 16:12:20.170772+00:00
[2025-03-06T16:12:25.200+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'spark_flow', 'load_exchangeRate_python_job', 'manual__2025-03-06T16:12:20.170772+00:00', '--job-id', '10', '--raw', '--subdir', 'DAGS_FOLDER/dag_***_spark.py', '--cfg-path', '/tmp/tmpj7ia79rr']
[2025-03-06T16:12:25.203+0000] {standard_task_runner.py:105} INFO - Job 10: Subtask load_exchangeRate_python_job
[2025-03-06T16:12:25.204+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=609) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-03-06T16:12:25.205+0000] {standard_task_runner.py:72} INFO - Started process 610 to run task
[2025-03-06T16:12:25.260+0000] {task_command.py:467} INFO - Running <TaskInstance: spark_flow.load_exchangeRate_python_job manual__2025-03-06T16:12:20.170772+00:00 [running]> on host 6dfac24f2f32
[2025-03-06T16:12:25.356+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='haonv' AIRFLOW_CTX_DAG_ID='spark_flow' AIRFLOW_CTX_TASK_ID='load_exchangeRate_python_job' AIRFLOW_CTX_EXECUTION_DATE='2025-03-06T16:12:20.170772+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-03-06T16:12:20.170772+00:00'
[2025-03-06T16:12:25.357+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-03-06T16:12:25.383+0000] {base.py:84} INFO - Retrieving connection 'spark-conn'
[2025-03-06T16:12:25.385+0000] {spark_submit.py:474} INFO - Spark-Submit cmd: spark-submit --master spark://172.18.0.4:7077 --name arrow-spark --deploy-mode client jobs/python/load_exchangeRateToDW.py
[2025-03-06T16:12:28.601+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:28 INFO SparkContext: Running Spark version 3.5.5
[2025-03-06T16:12:28.603+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:28 INFO SparkContext: OS info Linux, 5.15.167.4-microsoft-standard-WSL2, amd64
[2025-03-06T16:12:28.603+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:28 INFO SparkContext: Java version 17.0.14
[2025-03-06T16:12:28.652+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:28 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-03-06T16:12:28.732+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:28 INFO ResourceUtils: ==============================================================
[2025-03-06T16:12:28.733+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:28 INFO ResourceUtils: No custom resources configured for spark.driver.
[2025-03-06T16:12:28.734+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:28 INFO ResourceUtils: ==============================================================
[2025-03-06T16:12:28.734+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:28 INFO SparkContext: Submitted application: ExchangeRateToDW
[2025-03-06T16:12:28.752+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:28 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2025-03-06T16:12:28.760+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:28 INFO ResourceProfile: Limiting resource is cpu
[2025-03-06T16:12:28.761+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:28 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2025-03-06T16:12:28.806+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:28 INFO SecurityManager: Changing view acls to: ***
[2025-03-06T16:12:28.807+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:28 INFO SecurityManager: Changing modify acls to: ***
[2025-03-06T16:12:28.808+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:28 INFO SecurityManager: Changing view acls groups to:
[2025-03-06T16:12:28.808+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:28 INFO SecurityManager: Changing modify acls groups to:
[2025-03-06T16:12:28.809+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:28 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: ***; groups with view permissions: EMPTY; users with modify permissions: ***; groups with modify permissions: EMPTY
[2025-03-06T16:12:29.055+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO Utils: Successfully started service 'sparkDriver' on port 36963.
[2025-03-06T16:12:29.078+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO SparkEnv: Registering MapOutputTracker
[2025-03-06T16:12:29.110+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO SparkEnv: Registering BlockManagerMaster
[2025-03-06T16:12:29.123+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2025-03-06T16:12:29.124+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2025-03-06T16:12:29.129+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2025-03-06T16:12:29.148+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-e1ed4445-5f90-43cf-8bf6-8db02dc087aa
[2025-03-06T16:12:29.162+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2025-03-06T16:12:29.177+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO SparkEnv: Registering OutputCommitCoordinator
[2025-03-06T16:12:29.292+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2025-03-06T16:12:29.364+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2025-03-06T16:12:29.405+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 ERROR SparkContext: Failed to add /opt/bitnami/spark/jars/postgresql-42.7.5.jar to Spark environment
[2025-03-06T16:12:29.406+0000] {spark_submit.py:641} INFO - java.io.FileNotFoundException: Jar /opt/bitnami/spark/jars/postgresql-42.7.5.jar not found
[2025-03-06T16:12:29.407+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.addLocalJarFile$1(SparkContext.scala:2095)
[2025-03-06T16:12:29.408+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.addJar(SparkContext.scala:2151)
[2025-03-06T16:12:29.408+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.$anonfun$new$15(SparkContext.scala:521)
[2025-03-06T16:12:29.409+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.$anonfun$new$15$adapted(SparkContext.scala:521)
[2025-03-06T16:12:29.410+0000] {spark_submit.py:641} INFO - at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2025-03-06T16:12:29.411+0000] {spark_submit.py:641} INFO - at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2025-03-06T16:12:29.412+0000] {spark_submit.py:641} INFO - at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2025-03-06T16:12:29.412+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.<init>(SparkContext.scala:521)
[2025-03-06T16:12:29.413+0000] {spark_submit.py:641} INFO - at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
[2025-03-06T16:12:29.414+0000] {spark_submit.py:641} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2025-03-06T16:12:29.414+0000] {spark_submit.py:641} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[2025-03-06T16:12:29.415+0000] {spark_submit.py:641} INFO - at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2025-03-06T16:12:29.416+0000] {spark_submit.py:641} INFO - at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
[2025-03-06T16:12:29.416+0000] {spark_submit.py:641} INFO - at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
[2025-03-06T16:12:29.417+0000] {spark_submit.py:641} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
[2025-03-06T16:12:29.418+0000] {spark_submit.py:641} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2025-03-06T16:12:29.419+0000] {spark_submit.py:641} INFO - at py4j.Gateway.invoke(Gateway.java:238)
[2025-03-06T16:12:29.419+0000] {spark_submit.py:641} INFO - at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
[2025-03-06T16:12:29.420+0000] {spark_submit.py:641} INFO - at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
[2025-03-06T16:12:29.421+0000] {spark_submit.py:641} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2025-03-06T16:12:29.421+0000] {spark_submit.py:641} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2025-03-06T16:12:29.422+0000] {spark_submit.py:641} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2025-03-06T16:12:29.500+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://172.18.0.4:7077...
[2025-03-06T16:12:29.538+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO TransportClientFactory: Successfully created connection to /172.18.0.4:7077 after 21 ms (0 ms spent in bootstraps)
[2025-03-06T16:12:29.620+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20250306161229-0002
[2025-03-06T16:12:29.624+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20250306161229-0002/0 on worker-20250306160751-172.18.0.5-46537 (172.18.0.5:46537) with 2 core(s)
[2025-03-06T16:12:29.628+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO StandaloneSchedulerBackend: Granted executor ID app-20250306161229-0002/0 on hostPort 172.18.0.5:46537 with 2 core(s), 1024.0 MiB RAM
[2025-03-06T16:12:29.629+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20250306161229-0002/1 on worker-20250306160752-172.18.0.6-39231 (172.18.0.6:39231) with 2 core(s)
[2025-03-06T16:12:29.630+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO StandaloneSchedulerBackend: Granted executor ID app-20250306161229-0002/1 on hostPort 172.18.0.6:39231 with 2 core(s), 1024.0 MiB RAM
[2025-03-06T16:12:29.632+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 35799.
[2025-03-06T16:12:29.633+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO NettyBlockTransferService: Server created on 6dfac24f2f32:35799
[2025-03-06T16:12:29.635+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2025-03-06T16:12:29.640+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 6dfac24f2f32, 35799, None)
[2025-03-06T16:12:29.647+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO BlockManagerMasterEndpoint: Registering block manager 6dfac24f2f32:35799 with 434.4 MiB RAM, BlockManagerId(driver, 6dfac24f2f32, 35799, None)
[2025-03-06T16:12:29.650+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 6dfac24f2f32, 35799, None)
[2025-03-06T16:12:29.652+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 6dfac24f2f32, 35799, None)
[2025-03-06T16:12:29.671+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20250306161229-0002/0 is now RUNNING
[2025-03-06T16:12:29.673+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20250306161229-0002/1 is now RUNNING
[2025-03-06T16:12:29.955+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:29 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2025-03-06T16:12:30.408+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:30 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2025-03-06T16:12:30.421+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:30 INFO SharedState: Warehouse path is 'file:/opt/***/spark-warehouse'.
[2025-03-06T16:12:32.259+0000] {spark_submit.py:641} INFO - Traceback (most recent call last):
[2025-03-06T16:12:32.263+0000] {spark_submit.py:641} INFO - File "/opt/***/jobs/python/load_exchangeRateToDW.py", line 66, in <module>
[2025-03-06T16:12:32.274+0000] {spark_submit.py:641} INFO - .load()
[2025-03-06T16:12:32.275+0000] {spark_submit.py:641} INFO - ^^^^^^
[2025-03-06T16:12:32.276+0000] {spark_submit.py:641} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/streaming/readwriter.py", line 304, in load
[2025-03-06T16:12:32.277+0000] {spark_submit.py:641} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1322, in __call__
[2025-03-06T16:12:32.278+0000] {spark_submit.py:641} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/pyspark.zip/pyspark/errors/exceptions/captured.py", line 185, in deco
[2025-03-06T16:12:32.278+0000] {spark_submit.py:641} INFO - pyspark.errors.exceptions.captured.AnalysisException: Failed to find data source: kafka. Please deploy the application as per the deployment section of Structured Streaming + Kafka Integration Guide.
[2025-03-06T16:12:32.374+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:32 INFO SparkContext: Invoking stop() from shutdown hook
[2025-03-06T16:12:32.375+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:32 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2025-03-06T16:12:32.395+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:32 INFO SparkUI: Stopped Spark web UI at http://6dfac24f2f32:4040
[2025-03-06T16:12:32.401+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:32 INFO StandaloneSchedulerBackend: Shutting down all executors
[2025-03-06T16:12:32.406+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:32 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Asking each executor to shut down
[2025-03-06T16:12:32.429+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:32 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2025-03-06T16:12:32.446+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:32 INFO MemoryStore: MemoryStore cleared
[2025-03-06T16:12:32.447+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:32 INFO BlockManager: BlockManager stopped
[2025-03-06T16:12:32.456+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:32 INFO BlockManagerMaster: BlockManagerMaster stopped
[2025-03-06T16:12:32.459+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:32 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2025-03-06T16:12:32.468+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:32 INFO SparkContext: Successfully stopped SparkContext
[2025-03-06T16:12:32.468+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:32 INFO ShutdownHookManager: Shutdown hook called
[2025-03-06T16:12:32.469+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:32 INFO ShutdownHookManager: Deleting directory /tmp/spark-ba7f46fe-0a9c-421f-acd1-87d2f882fd1b
[2025-03-06T16:12:32.473+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:32 INFO ShutdownHookManager: Deleting directory /tmp/spark-1689eb23-0e11-402e-9ed3-2ad47b1b8f83/pyspark-f3627f71-8d66-424c-9ac2-365535d26746
[2025-03-06T16:12:32.477+0000] {spark_submit.py:641} INFO - 25/03/06 16:12:32 INFO ShutdownHookManager: Deleting directory /tmp/spark-1689eb23-0e11-402e-9ed3-2ad47b1b8f83
[2025-03-06T16:12:32.522+0000] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 177, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 567, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://172.18.0.4:7077 --name arrow-spark --deploy-mode client jobs/python/load_exchangeRateToDW.py. Error code is: 1.
[2025-03-06T16:12:32.526+0000] {taskinstance.py:1226} INFO - Marking task as FAILED. dag_id=spark_flow, task_id=load_exchangeRate_python_job, run_id=manual__2025-03-06T16:12:20.170772+00:00, execution_date=20250306T161220, start_date=20250306T161225, end_date=20250306T161232
[2025-03-06T16:12:32.562+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-03-06T16:12:32.562+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 10 for task load_exchangeRate_python_job (Cannot execute: spark-submit --master spark://172.18.0.4:7077 --name arrow-spark --deploy-mode client jobs/python/load_exchangeRateToDW.py. Error code is: 1.; 610)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 177, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 567, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://172.18.0.4:7077 --name arrow-spark --deploy-mode client jobs/python/load_exchangeRateToDW.py. Error code is: 1.
[2025-03-06T16:12:32.585+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-03-06T16:12:32.613+0000] {taskinstance.py:3901} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-03-06T16:12:32.619+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
